crawler/config/centres.py

```diff
@@ -239,3 +263,42 @@
         CENTRE_KEY_SKIP_UNCONSOLIDATED_SURVEILLANCE_FILES: False,
     },
 ]
+
+
+def get_centres_config(config: Config, data_source: str = "") -> List[CentreConf]:
+    """Get the centres config from MongoDB. If MongoDB does not contain any centres config, it will become populated
+    with the values in the app config for centres.
+
+    Arguments:
+        config {Config}: The configuration object for the whole application.
+        data_source {str}: The data source filter to apply to centre configs, or None to apply no filter.
+
+    Return:
+        List[CentreConf]: A List of CentreConf from MongoDB matching the given data source.
+    """
+    with create_mongo_client(config) as client:
+        db = get_mongo_db(config, client)
+
+        centres_collection_exists = collection_exists(db, COLLECTION_CENTRES)
+        centres_collection = get_mongo_collection(db, COLLECTION_CENTRES)
+
+        if not centres_collection_exists:
+            # Populate the centres collection from the config values
+            create_index(centres_collection, FIELD_CENTRE_NAME, unique=True)
+            populate_collection(centres_collection, config.CENTRES, FIELD_CENTRE_NAME)  # type: ignore
+
+        # Get the centres collection from MongoDB
```

Comment:
and how does this way of getting the collection differ from the method above?

Reply:
The getter you commented on above gets a reference to the collection.  The `if` block populates it with the config file if it didn't already exist.  Then this code here asks the collection reference to get all the documents it contains and casts them to `CentreConf`.  This means we always get the `CentreConf` objects the same way, whether we just inserted them into MongoDB or not.